src/disk.hpp

```diff
@@ -219,6 +216,14 @@ struct FileDisk {
                 std::this_thread::sleep_for(5min);
             }
         } while (amtwritten != length);
+    #ifdef _WIN32
+        // Flush Windows Cache after few cycles
+        flush_cyclecount += 1;
+        if (flush_cyclecount >= flush_cyclelimit){
```

Comment:
This makes more sense. However, if this turns out to give some people worse performance, they really should have a configuration option to disable it or change this limit.

The limit should probably be specified in number-of-bytes written to the file too.